### ISSUE TITLE
HTML5 fullscreen api not working in webview-tag

### ISSUE THREAD

**Issue by thyrrestrup** [atom/electron#1503]
Hi 
I just pulled the master and tried the new fullscreen feature.
It works perfect as long as the video is not embedded inside a webview.
Is it possible to fix this?

-Tobias

**Comment by gliurn** [comment]
+1

**Comment by Tribex** [comment]
+1 here as well. :)

**Comment by jesperunna** [comment]
+1 !!!

**Comment by deepak1556** [comment]
can anyone provide a sample snippet of how you are embedding the player in webview ? with object based embed using pepper flash player fullscreen works with webview. So i believe this is just an issue of missing `allowfullscreen` attribute on webview tag creation for html5.
 
```html
  <object width="500" height="500">
          <param name="allowfullscreen" value="true" />
          <param name="allowscriptaccess" value="always" />
          <param name="movie" value="https://vimeo.com/moogaloop.swf?clip_id=126725607&amp;server=vimeo.com&amp;color=00adef&amp;fullscreen=1" />
          <embed src="https://vimeo.com/moogaloop.swf?clip_id=126725607&amp;server=vimeo.com&amp;color=00adef&amp;fullscreen=1"
              type="application/x-shockwave-flash" allowfullscreen="true" allowscriptaccess="always" width="500" height="500"></embed>
      </object>
```

**Comment by thyrrestrup** [comment]
Here is a sample that shows the problem when embedding youtube and vimeo videos:
`index.html`
```html
<!DOCTYPE html>
<html>
    <head>
        <title>Webview</title>
        <script>
            // Here to make sure the webview shows up...
        </script>
        <style>
            .webview {
                position: absolute;
                top: 0;
                bottom: 0;
                left: 0;
                right: 0;
                margin: 0;
                padding: 0;
                outline: none !important;
            }
        </style>
    </head>
    <body>
        <webview src="./video.html" class="webview" allowfullscreen></webview>
    </body>
</html>
```
`video.html`
```html
<!DOCTYPE html>
<html>
    <head>
        <title>Fullscreen Video</title>
    </head>
    <body>
        <iframe width="640" height="360" src="https://www.youtube.com/embed/dIkHPxeslHQ" frameborder="0" allowfullscreen></iframe>
        <iframe width="640" height="360" src="https://player.vimeo.com/video/106724185?color=ffffff&portrait=0" frameborder="0" allowfullscreen></iframe>
    </body>
</html>
```
If `video.html` is loaded from the webview it does not work but if it's loaded directly into the browser-window it works beautifully.
When `video.html` is loaded through the webview and the fullscreen button is clicked nothing happens with the video but scroll stops working on the page. Same issue as before HTML5 fullscreen was implemented in the browser-window #1314.

-Tobias

**Comment by thyrrestrup** [comment]
@deepak1556 - I've just tested your pull request and it fixes the scenario in the sample I provided. Thanks
But it doesn't work on the following site if it's implemented in webview:
https://www.youtube.com/user/StateofGreenDK
It does however work if it's not in a webview.

**Comment by deepak1556** [comment]
@thyrrestrup fullscreen seems to work there too in webview, i am not able to reproduce the crash.. AM i missing something here ?

```
<!DOCTYPE html>
<html>
    <head>
        <title>Webview</title>
        <script>
            // Here to make sure the webview shows up...
        </script>
        <style>
            .webview {
                position: absolute;
                top: 0;
                bottom: 0;
                left: 0;
                right: 0;
                margin: 0;
                padding: 0;
                outline: none !important;
            }
        </style>
    </head>
    <body>
        <webview src="https://www.youtube.com/user/StateofGreenDK" class="webview"></webview>
    </body>
</html>
```

**Comment by thyrrestrup** [comment]
@deepak1556 - The example you gave works flawlessly - I've narrowed it down to this sample:
`index.html`
```html
<!DOCTYPE html>
<html>
    <head>
        <title>Webview</title>
        <script>
            onload = function() {
                var content = document.getElementById('content');
                content.addEventListener('new-window', function(e) {
                    content.src = e.url;
                });
            };
        </script>
        <style>
            .webview {
                position: absolute;
                top: 0;
                bottom: 0;
                left: 0;
                right: 0;
                margin: 0;
                padding: 0;
                outline: none !important;
            }
        </style>
    </head>
    <body>
        <webview id="content" src="./link.html" class="webview" plugins></webview>
    </body>
</html>
```
`link.html`
```html
<!DOCTYPE html>
<html>
    <head>
        <title>Link</title>
    </head>
    <body>
        <a href="http://www.youtube.com/user/StateofGreenDK" target="_blank">YouTube</a>
    </body>
</html>
```
I guess something weird happens when changing the src of the webview...

**Comment by deepak1556** [comment]
Ah thanks for the test case, @zcbenz in the above scenario the intermediate page loaded from `link.html` doesnt get recorded on [`dying_render_process_`](https://github.com/atom/electron/blob/05ae1960d1e421cf057a960b0526890665983c47/atom/browser/atom_browser_client.cc#L131) so it leads to a werid scenario in `UpdateGuestProcessID`. For example,

   `Page Transition`  |  `Old Id` | `New Id`
   ----------------------   |  ---------   | ----------
  index.html load      |      6        |    7
  link.html load         |      7        |    8
  onclick link             |     7         |   9

the solution i can think of is have the `webview_info_map_` as append only [assuming ids are incremental](https://github.com/atom/electron/blob/05ae1960d1e421cf057a960b0526890665983c47/atom/browser/web_view_manager.cc#L47) or find another way to record dying render process. Thoughts ?

**Comment by deepak1556** [comment]
@thyrrestrup fullscreen will work as long as the guest web contents can be associated to a window :)

**Comment by thyrrestrup** [comment]
@deepak1556 - Thanks for tracking this down.

**Comment by thyrrestrup** [comment]
@deepak1556 - Do you think we should close this and open a new issue with your findings?

**Comment by deepak1556** [comment]
We can close this once the PR to fix fullscreen in webview is merged :) but for the other scenario yup a new issue sounds good.

**Comment by thyrrestrup** [comment]
@deepak1556 - I've come across a couple of issues with the HTML5 fullscreen implementation.

1. If the browser window is already in fullscreen nothing happens when pushing the fullscreen button on the video. If the window is then taken out of fullscreen weird things happen with the video.
2. If the html contains multiple webview's all of them are included in the fullscreen (see attached image) - toolbar is first webview and video is second webview.

Is it possible to create a completely new window for the video content and assign it a window level above the fullscreen browser window? and also to only include the webview that contains the video?

![screen shot 2015-05-15 at 09 53 20](https://cloud.githubusercontent.com/assets/415042/7649132/557591e6-faea-11e4-85bf-7d800c3da3ca.png)
